src/lava/magma/core/model/py/ports.py

```diff
@@ -513,6 +513,14 @@ def write(
         """
         pass
 
+    # TODO: (PP) This should be optimized by a proper CSPSendPort wait
+    def wait(self):
+        """Method which block execution until (all) previous write requests have
+         been placed at the receiver. Currently, achieved by a simple read(),
+         which ensures the writes have been finished. """
+
+        self.read()
+
```

Comment:
Doesn't it make sense to call self.wait() at the end of self.write(...) routine? Is there any reason why we would want the non-deterministic behavior?